File: apple-file-events/js/src/main/scala/com/swoval/files/apple/FileEvent.scala
package com.swoval.files.apple

import com.swoval.files.apple.Flags.Event._

//remove if not needed
import scala.collection.JavaConversions._

class FileEvent(val fileName: String, var flags: Int)
    extends com.swoval.files.apple.Event {

  private def hasFlags(f: Int): Boolean = (flags & f) != 0

  def isModified(): Boolean = hasFlags(ItemInodeMetaMod | ItemModified)

  def isNewFile(): Boolean =
    isCreated && !isModified && !isRemoved && !isTouched

  def isCreated(): Boolean = hasFlags(ItemCreated)

  def isRemoved(): Boolean = hasFlags(ItemRemoved)

  def isTouched(): Boolean = hasFlags(ItemInodeMetaMod)

  override def toString(): String =
    "FileEvent(" + fileName + ", " + Flags.Event.flags(this) +
      ")"

}

File: apple-file-events/jvm/src/main/java/com/swoval/files/apple/FileEvent.java
package com.swoval.files.apple;

import static com.swoval.files.apple.Flags.Event.ItemCreated;
import static com.swoval.files.apple.Flags.Event.ItemInodeMetaMod;
import static com.swoval.files.apple.Flags.Event.ItemModified;
import static com.swoval.files.apple.Flags.Event.ItemRemoved;

public class FileEvent implements com.swoval.files.apple.Event {

    public final String fileName;
    private final int flags;

    public FileEvent(final String fileName, final int flags) {
        this.fileName = fileName;
        this.flags = flags;
    }

    private boolean hasFlags(int f) {
        return (flags & f) != 0;
    }

    public boolean isModified() {
        return hasFlags(ItemInodeMetaMod | ItemModified);
    }

    public boolean isNewFile() {
        return isCreated() && !isModified() && !isRemoved() && !isTouched();
    }

    public boolean isCreated() {
        return hasFlags(ItemCreated);
    }

    public boolean isRemoved() {
        return hasFlags(ItemRemoved);
    }

    public boolean isTouched() {
        return hasFlags(ItemInodeMetaMod);
    }

    @Override
    public String toString() {
        return "FileEvent(" + fileName + ", " + Flags.Event.flags(this) + ")";
    }

    @Override
    public int flags() {
        return flags;
    }
}

